test/GPUActivityAgentTest.cpp

```diff
@@ -0,0 +1,434 @@
+        .WillByDefault(Return(GPU_FREQUENCY_CONTROL_MAX_IDX));
+    ON_CALL(*m_platform_io, agg_function(_))
+        .WillByDefault(Return(geopm::Agg::average));
+
+    m_freq_min = 0135000000.0;
+    m_freq_max = 1530000000.0;
+    ON_CALL(*m_platform_io, control_domain_type("GPU_CORE_FREQUENCY_MIN_CONTROL"))
+        .WillByDefault(Return(GEOPM_DOMAIN_GPU));
+    ON_CALL(*m_platform_io, control_domain_type("GPU_CORE_FREQUENCY_MAX_CONTROL"))
+        .WillByDefault(Return(GEOPM_DOMAIN_GPU));
+    ON_CALL(*m_platform_io, signal_domain_type("GPU_CORE_ACTIVITY"))
+        .WillByDefault(Return(GEOPM_DOMAIN_GPU));
+    ON_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MIN_AVAIL", GEOPM_DOMAIN_BOARD, 0))
+        .WillByDefault(Return(m_freq_min));
+    ON_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MAX_AVAIL", GEOPM_DOMAIN_BOARD, 0))
+        .WillByDefault(Return(m_freq_max));
+
+    ASSERT_LT(m_freq_min, 0.2e9);
+    ASSERT_LT(1.4e9, m_freq_max);
+
+    m_agent = geopm::make_unique<GPUActivityAgent>(*m_platform_io, *m_platform_topo);
+    m_num_policy = m_agent->policy_names().size();
+
+    m_default_policy = {m_freq_max, m_freq_min, NAN};
+
+    // leaf agent
+    m_agent->init(0, {}, false);
+}
+
+void GPUActivityAgentTest::TearDown()
+{
+
+}
+
+TEST_F(GPUActivityAgentTest, name)
+{
+    EXPECT_EQ("gpu_activity", m_agent->plugin_name());
+    EXPECT_NE("bad_string", m_agent->plugin_name());
+}
+
+TEST_F(GPUActivityAgentTest, validate_policy)
+{
+    //Called as part of validate
+    EXPECT_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MIN_AVAIL", _, _)).WillRepeatedly(
+                Return(m_freq_min));
+    EXPECT_CALL(*m_platform_io, read_signal("GPU_CORE_FREQUENCY_MAX_AVAIL", _, _)).WillRepeatedly(
+                Return(m_freq_max));
+    EXPECT_CALL(*m_platform_topo, num_domain(GEOPM_DOMAIN_BOARD)).WillRepeatedly(Return(M_NUM_BOARD));
+
+    const std::vector<double> empty(m_num_policy, NAN);
+    std::vector<double> policy;
+
+    // default policy is accepted
+    // load default policy
+    policy = m_default_policy;
+    m_agent->validate_policy(policy);
+    // validate policy is unmodified except Phi
+    ASSERT_EQ(m_num_policy, policy.size());
+    EXPECT_EQ(m_freq_max, policy[FREQ_MAX]);
+    EXPECT_EQ(m_freq_min, policy[FREQ_EFFICIENT]);
+    // Default value when NAN is passed is 0.5
+    EXPECT_EQ(0.5, policy[PHI]);
+
+    // all-NAN policy is accepted
+    // setup & load NAN policy
+    policy = empty;
+    m_agent->validate_policy(policy);
+    // validate policy defaults are applied
+    ASSERT_EQ(m_num_policy, policy.size());
+    EXPECT_EQ(m_freq_max, policy[FREQ_MAX]);
+    EXPECT_EQ((policy[FREQ_MAX] + m_freq_min) / 2, policy[FREQ_EFFICIENT]);
+    EXPECT_EQ(0.5, policy[PHI]);
+
+    // non-default policy is accepted
+    // setup & load policy
+    policy[FREQ_MAX] = m_freq_max;
+    policy[FREQ_EFFICIENT] = m_freq_max / 2;
+    policy[PHI] = 0.1;
+    m_agent->validate_policy(policy);
```

Comment:
Expect that it doesn't throw. Same applies to all calls of validate_policy(), really.